external-crates/move/crates/move-compiler/src/naming/translate.rs

```diff
@@ -1442,9 +1918,58 @@ fn exp(context: &mut Context, e: Box<E::Exp>) -> Box<N::Exp> {
                 }
             }
         }
+        EE::Name(ma @ sp!(_, E::ModuleAccess_::Variant(_, _)), etys_opt) => {
+            context
+                .env
+                .check_feature(FeatureGate::Enums, context.current_package, eloc);
+            match context.resolve_variant_name(eloc, "construction", ma, etys_opt) {
+                None => {
+                    assert!(context.env.has_errors());
+                    NE::UnresolvedError
+                }
+                Some((m, en, vn, tys_opt, dloc, field_info)) => {
+                    if !field_info.is_empty() {
+                        let msg =
+                            "Invalid variant instantiation. Non-empty variant instantiations \
```

Comment:
Wondering if "instantiation" is the right word here? Thoughts on maybe "construction"?

Reply:
Instantiation is consistent with the struct case.